wordpress-mobile/WordPress-Android
# ViewBinding: Remove synthetic accessors: Site Creation

Parent #14845 

This PR replaces synthetic accessors within two steps of the SiteCreation flow.

Details
- SiteCreationBaseFormFragment
	- Converted from java to kotlin
	- View binding takes place view in `onViewCreated`, so need to set a `binding` field
	- Added abstract fun`setBindingViewStubListener` to enable subclasses to inflate their own layout
	- Replace synthetic accessors with view binding

- SiteCreationDomainsFragment
	- Replaced synthetic accessors with view binding
	- Implemented `setBindingViewStubListener` to inflate layout into ViewStub

- SiteCreationPreviewFragment
	- Replaced synthetic accessors with view binding
	- Implemented `setBindingViewStubListener` to inflate layout into ViewStub
	- Nix animation when any required param is null (a.k.a. - happens if view binding are null)

- Layout files
	- Added `@+id\` to all includes
	- Extracted out layout-land `site_creation_preivew_screen` into layout-land `site_creation_preivew_screen_default` to match portrait/h480dp layouts. This enforces that we access the same `binding` fields throughout `SiteCreationPreviewFragment`

To Test
- Launch the app
- Make sure the device is using Portrait configuration
- Navigate to MySites 
- Choose the site selector to open the Sites List View
- Tap the + in the upper right hand corner
- Tap Create WordPress.com site
- Select a template (doesn't matter) and tap the Choose button
- (1) Note the Choose a domain view is shown
- (2) Test that the search feature is performing as expected
- Select a domain name and tap Create Site
- (3) Note the progress view is shown
- (4) Note the progress view transitions to a preview view once the site is created

- Repeat the test for Landscape configuration (the PixelC emulator works great)
## Regression Notes
1. Potential unintended areas of impact
Site Creation flow

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tested the site creation flow

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
